tslearn/docs/examples/classification/plot_shapelet_distances.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""
+Learning Shapelets: decision boundaries in 2D distance space
+============================================================
+
+This example illustrates the use of the "Learning Shapelets" method in order
+to learn a collection of shapelets that linearly separates the timeseries.
+In this example, we will extract two shapelets which are then used to
+transform our input time series in a two-dimensional space, which is called
+the shapelet-transform space in the related literature. Moreover, we plot the
+decision boundaries of our classifier for each of the different classes.
+
+More information on the method can be found at:
+http://fs.ismll.de/publicspace/LearningShapelets/.
+"""
+
+# Author: Gilles Vandewiele
+# License: BSD 3 clause
+
+import numpy
+from matplotlib import cm
+import matplotlib.pyplot as plt
+
+from tslearn.datasets import CachedDatasets
+from tslearn.preprocessing import TimeSeriesScalerMinMax
+from tslearn.shapelets import LearningShapelets
+from tensorflow.keras.optimizers import SGD
+
+# Set a seed to ensure determinism
+numpy.random.seed(42)
+
+# Load the Trace dataset
+X_train, y_train, _, _ = CachedDatasets().load_dataset("Trace")
+
+# Normalize the time series
+X_train = TimeSeriesScalerMinMax().fit_transform(X_train)
+
+# Get statistics of the dataset
+n_ts, ts_sz = X_train.shape[:2]
+n_classes = len(set(y_train))
+
+# We will extract 2 shapelets and align them with the time series
+shapelet_sizes = {20: 2}
+
+# Define the model and fit it using the training data
+shp_clf = LearningShapelets(n_shapelets_per_size=shapelet_sizes,
+                            weight_regularizer=0.0001,
+                            optimizer=SGD(lr=0.9),
+                            max_iter=1000,
+                            verbose=0,
+                            scale=False,
+                            random_state=42)
+shp_clf.fit(X_train, y_train)
+
+# We will plot our distances in a 2D space
+distances = shp_clf.transform(X_train).reshape((-1, 2))
+weights, biases = shp_clf.model_.layers[-1].get_weights()
```

Comment:
Is there a reason for not using get weights from the LearningShapelet instance directly here ?

Reply:
I actually have to remove that code since I do not use that no longer (I just use meshgrid and use the predict function to draw the decision boundaries)

Reply:
Ok no sorry I do use them to make my predictions, let me change that.

Reply:
Fixed! Replaced by `shp_clf.get_weights('classification')`